core/ledger/kvledger/kv_ledger.go

```diff
@@ -187,6 +190,35 @@ func (l *kvLedger) Prune(policy commonledger.PrunePolicy) error {
 	return errors.New("Not yet implemented")
 }
 
+// GetPrivateSimulationResults returns a private simulation results for a given transaction id
+func (l *kvLedger) GetPrivateSimulationResults(txId string) (commonledger.ResultsIterator, error) {
+	return l.transientStore.GetTxPrivateRWSetByTxId(txId)
+}
+
+// PutPrivateSimulationResults stores a private simulation results generated by an endorser as a
+// result of transaction simulation
+func (l *kvLedger) PutPrivateSimulationResults(txId string, endorserId string, privateSimulationResults []byte) error {
+	// Endorser should return the readLock only after storing the private write set in the
+	// transient store.
+	blockchainInfo, err := l.blockStore.GetBlockchainInfo()
+	if err != nil {
+		return err
+	}
+	return l.transientStore.Persist(txId, endorserId, blockchainInfo.Height, privateSimulationResults)
```

Comment:
Obtaining the block height in this manner is not correct... We would need to take this info in the API itself. E.g., Gossip may be calling this API for tran simulated at some other peer.